Fix `count_ratio` estimation

drodarie opened the pull request:
## List which issues this resolves:
Fixes #822

Helveg reviewed (changes requested):
It looks like you've interleaved 2 tests in 1, including duplicating the asserts in subtests. This can confound the test failures. Can you separate it? (Reviewed on my phone, maybe they really do belong in there)

drodarie commented:
I split the tests on guess and renamed them to make what is tested explicit. 
For the cascade test, I wonder if the different indicator should be split since the change on one should not affect the other?

Helveg reviewed (approved):
Ok! For peace of mind: what exactly is being tested here? Do you really need to re-assert all test cases of the old test on this new changeset?

> I split the tests on guess and renamed them to make what is tested explicit.
For the cascade test, I wonder if the different indicator should be split since the change on one should not affect the other?

The code is unclear, I created a bunch of cases for all the edge cases an indicator could face. It seems unlikely that you need to test more than 2 cases for this: does the guess work for regular cells, and for entities, or per voxel. The rest of the cases is general to indicators, and already tested in the existing tests, imho.

Helveg commented:
@drodarie If you're not planning to change the tests any further, feel free to merge this! :)

drodarie commented:
Sorry for the delay, based on your feedback, I would like to refactor and extend the tests.

Helveg commented on bsb/placement/indicator.py:
```diff
@@ -113,16 +112,19 @@ def guess(self, chunk=None, voxels=None):
                         rel_pl_density * density_ratio, chunk
                     )
+                elif rel_pl_density_key is not None:
+                    estimate = rel_ind.guess(chunk, voxels) * density_ratio
```
```suggestion
                    # Use the relation's `guess` to guess according to the relation's density key
                    estimate = rel_ind.guess(chunk, voxels) * density_ratio
```
